app/src/main/java/com/github/scribeWizTeam/scribewiz/Activities/NotesDisplayedActivity.kt

```diff
@@ -135,4 +138,52 @@ class NotesDisplayedActivity : AppCompatActivity() {
             shouldSetPosition = false
         }
     }
+
+    // This function is used to edit the note at the current tick position.
+    // It takes in the filePassed parameter which is a URI string pointing to the input musicXML file
+    // and the newNote parameter which is the note to replace the existing note with.
+    public fun editNote(filePassed: String, newNote: String): File {
+
+        // Convert the filePassed URI string to a file and create a temporary file to store the modified musicXML
+        val inputFileUri = Uri.parse(filePassed)
+        val inputFile = Companion.createTempFileFromUri(this, inputFileUri)
+        val outputFile = File.createTempFile("temp_musicxml_modified", ".xml", cacheDir)
+
+        // Get the current tick position and convert it to a note location in the input musicXML file
+        val tickPosition = _viewModel.currentTickPosition.value
+        val noteLocation = Editor.convertTicksToNoteLocation(inputFile, tickPosition!!)
+
+        // Edit the note in the input musicXML file and write the modified content to the output file
+        Editor.editNoteInMusicXML(outputFile, inputFile, noteLocation, newNote)
+
+        // Delete the input file since it's no longer needed and uses up storage space
+        inputFile.delete()
+
+        // Open the modified musicXML file in the app (refreshes the view)
+        openFile(Uri.fromFile(outputFile))
+
+        // Return the output file
+        return outputFile
+    }
+
+    companion object {
+        // A helper function that creates a temporary file from a content URI
+        fun createTempFileFromUri(notesDisplayedActivity: NotesDisplayedActivity, uri: Uri): File {
+            // Create a temporary file with a prefix "temp_musicxml" and a suffix ".xml" in the cache directory of the activity
+            val tempFile = File.createTempFile("temp_musicxml", ".xml", notesDisplayedActivity.cacheDir)
```

Comment:
Consider storing the strings in constants

Reply:
Thank you for this advice which i need to do better on generally, I change this on a later revision.